From 5fbb288022e2758051a79314b644474fe9e679b9 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:36:27 -0400
Subject: [block] blk-mq: remember to start timeout handler for direct queue

Message-id: <1402670238-13916-100-git-send-email-snitzer@redhat.com>
Patchwork-id: 83836
O-Subject: [RHEL7.1 PATCH 099/150] blk-mq: remember to start timeout handler for direct queue
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit feff6894128e4bb3c903ce247edc5bd3a1b09fc9
Author: Jens Axboe <axboe@fb.com>
Date:   Fri May 30 15:42:56 2014 -0600

    blk-mq: remember to start timeout handler for direct queue

    Commit 07068d5b8e added a direct-to-hw-queue mode, but this mode
    needs to remember to add the request timeout handler as well.
    Without it, we don't track timeouts for these requests.

    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 71f564e..5443aea 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -1217,6 +1217,7 @@ static void blk_mq_make_request(struct request_queue *q, struct bio *bio)
 
   blk_mq_bio_to_request(rq, bio);
   blk_mq_start_request(rq, true);
+  blk_add_timer(rq);
 
   /*
    * For OK queue, we are done. For error, kill it. Any other
-- 
1.7.1